zmoney/Services/CacheService/Extensions/Transaction+Storable.swift

```diff
@@ -46,9 +46,13 @@ extension Transaction: Storable {
         transactionEntity.incomeBankID = self.incomeBankID
         transactionEntity.outcomeBankID = self.outcomeBankID
         transactionEntity.reminderMaker = self.reminderMarker
-        transactionEntity.categories = self.categories
-        transactionEntity.incomeTransactionInstrument = self.incomeTransactionInstrument
-        transactionEntity.outcomeTransactionInstrument = self.outcomeTransactionInstrument
+        if let categories = self.categories {
+            for category in categories {
+                transactionEntity.addToCategories(category.makeEntity(context: context))
+            }
+        }
```

Comment:
Здесь чтобы pyramid of doom не создавать, можно воспользоваться 

 ```suggestion
        categories?.forEach { category in 
            transactionEntity.addToCategories(category.makeEntity(context: context))
        }
```

либо

 ```suggestion
        for category in categories ?? [] { 
            transactionEntity.addToCategories(category.makeEntity(context: context))
        }
```